contrib/fedora/rpm/firewalld.spec

```diff
@@ -0,0 +1,412 @@
+# SPEC file to build firewalld for testing. It aims to be the template for
+# RHEL/Fedora.
+#
+# This spec file is not directly usable. Instead, it contains __PLACEHOLDERS__ that
+# are adjusted by the accompanying "build.sh" script.
+
+%global version __VERSION__
+%global release_version __RELEASE_VERSION__
+%global snapshot __SNAPSHOT__
+%global git_sha __COMMIT__
+
+%if "x%{?snapshot}" != "x"
+%global snapshot_dot .%{snapshot}
+%endif
+%if "x%{?git_sha}" != "x"
+%global git_sha_dot .%{git_sha}
+%endif
+
+%global snap %{?snapshot_dot}%{?git_sha_dot}
+
+%if 0%{?fedora}
+%bcond_without fedora_variant
+%else
+%bcond_with    fedora_variant
+%endif
+
+%if 0%{?rhel}
+%bcond_with    firewalld_test
+%else
+%bcond_without firewalld_test
+%endif
+
+Summary: A firewall daemon with D-Bus interface providing a dynamic firewall
+Name: firewalld
+Version: %{version}
+Release: %{release_version}%{?snap}%{?dist}
+URL:     http://firewalld.org
+License: GPL-2.0-or-later
+Source0: https://github.com/firewalld/firewalld/releases/download/v%{version}/firewalld-%{version}.tar.bz2
+Source1: FedoraServer.xml
+Source2: FedoraWorkstation.xml
+Source3: org.fedoraproject.FirewallD1.desktop.rules.choice
+
+# The patches starting from 0001+ are downstream-only patches that apply
+# on both Fedora and RHEL.
+# They always apply, also after are rebase to a newer release tarball.
+#Patch0001: 0001-some.patch
+
+%if 0%{?fedora}
+# The patches starting from 1001+ are downstream-only patches only for Fedora.
+# They always apply, also after are rebase to a newer release tarball.
+Patch1001: 1001-fedora-only-MDNS-default.patch
+%else
+Source1001: 1001-fedora-only-MDNS-default.patch
+%endif
+
+%if 0%{?rhel}
+# The patches starting from 2001+ are downstream-only patches only for RHEL.
+# They always apply, also after are rebase to a newer release tarball.
+Patch2001: 2001-RHEL-only-Add-cockpit-by-default-to-some-zones.patch
+%else
+Source2001: 2001-RHEL-only-Add-cockpit-by-default-to-some-zones.patch
+%endif
+
```

Comment:
Unfortunately, we also often backport features. Maybe we need a placeholder for that as well. In theory it could go in the 9001+ range; so maybe it's not necessary.

Reply:
When doing backports, I don't think there needs to be a distinction between bugfix and features. In general, I find this distinction often unclear. It's just another patch. 

I would add them in 9001+ range. Also, because it's slightly more cumbersome, to add a patch at 3001 when there are already patches on 9001+. It's easier to just append new patches to the list at 9001+.

But sure, a range for "features" would work too...